pinc/Activity.inc

```diff
@@ -18,7 +18,91 @@ $ACCESS_CRITERIA = [
 
 // XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
 
-$Activity_for_id_ = [];
+/*
+ * A singleton container class for accessing Activity and inherited objects
+ * (Stages, Rounds, Pools).
+ *
+ * The Activity constructor adds the constructed object to this class and
+ * because Stage, Round, and Pool classes inherit from the Activity class,
+ * those objects are accessible from this class too. For code readability,
+ * those classes have accessor classes of their own (Stages, Rounds, Pools)
+ * that derive from Activities and specify which objects to access.
+ */
+class Activities
+{
```

Comment:
(I put this comment in the wrong place earlier)
It would be simpler not to extend Activities but have Activities, Stages, Pools and Rounds to be independent classes. Then when the different activities are created each one will get the appropriate activities set in it: Stages will get all the Rounds and Pools, Activities will get everything. When you create a Round it creates its parent Stage which will put the Round into its Stages class and so on.
The classes would be the same shape so you could define them with a trait like
```php
trait StatObj
{
    private static $objects = [];

    public static function set($object)
    {
        self::$objects[$object->id] = $object;
    }

    public static function get_all()
    {
        return self::$objects;
    }

    public static function get_by_id($id)
    {
        return self::$objects[$id] ?? null;
    }
}

class Activities
{
    use StatObj;
}
```

Reply:
It looks like get_ids() is also needed, which could be put in the trait
```php
    public static function get_ids()
    {
        return array_keys(self::$objects);
    }
```

Reply:
> It would be simpler not to extend Activities but have Activities, Stages, Pools and Rounds to be independent classes.

Perhaps, but I rather like that as-is the "collection" classes have inheritance like the objects they contain (Activity, Stage, Round, Pool). The current way also is more efficient that we only actually map the object once at the base level -- Activities -- rather than duplicate it everywhere. This should be very minor though as we are storing a reference to the same object, not a new object every time (this was true in the original implementation too).

 I'll have to give it more thought.